crates/sui-core/src/signature_verifier.rs

```diff
@@ -92,12 +92,16 @@ pub struct SignatureVerifier {
     certificate_cache: VerifiedDigestCache<CertificateDigest>,
     signed_data_cache: VerifiedDigestCache<SenderSignedDataDigest>,
 
-    /// Map from kid (key id) to the fetched OAuthProviderContent for that key.
+    /// Map from (kid, iss) to the fetched JWK for that key.
     /// We use an immutable data structure because verification of ZKLogins may be slow, so we
     /// don't want to pass a reference to the map to the verify method, since that would lead to a
     /// lengthy critical section. Instead, we use an immutable data structure which can be cloned
     /// very cheaply.
-    oauth_provider_jwk: RwLock<ImHashMap<String, OAuthProviderContent>>,
+    // todo(joyqvq): check what to use here since im is not apache.
+    oauth_provider_jwk: RwLock<HashMap<(String, String), JWK>>,
+
+    supported_providers: RwLock<Vec<OIDCProvider>>,
+    zklogin_env: RwLock<ZkLoginEnv>,
```

Comment:
hopefully we can remove this and just look at the protocol config directly, but if not, please put these under a single lock so that atomicity is guaranteed

Reply:
combined in one rw lock, and im loading this from chain identifier